### ISSUE TITLE
SOTA results using Llama 3.1-8B-Instruct (SFT LoRa-bnb-4bit)

### ISSUE THREAD

**Issue by abdallaharioua** [yzjiao/On-Demand-IE#2]
Hello,

Just wanted to share my results on finetuning Llama3.1-8B-Instruct (4bit bnb, training took 1h30 on 2xA100 80GB, 32 epochs). Many thanks for the scripts, they worked very well, and I hope this can be of value. For the record, I attached the reported results from the paper.
PS: I didn't compare directly with Llama 3.1-8b-Inst, maybe the finetuning is not adding any perf, let me know if you manage to do it.

#### Header Evaluation (F1 Score)
<table>
  <tr>
    <th rowspan="2">Model</th>
    <th rowspan="2">Overall</th>
    <th colspan="2">Source Type</th>
    <th colspan="2">Category</th>
    <th colspan="3">Difficulty</th>
  </tr>
  <tr>
    <th>Generate</th>
    <th>Retrieve</th>
    <th>Fixed Header</th>
    <th>Open Header</th>
    <th>Hard</th>
    <th>Easy</th>
    <th>Medium</th>
  </tr>
  <tr>
    <td>ODIE-8B-4Bit-Llama3.1-inst</td>
    <td>74.91</td>
    <td>73.05</td>
    <td>75.52</td>
    <td>82.16</td>
    <td>58.25</td>
    <td>73.77</td>
    <td>77.79</td>
    <td>73.23</td>
  </tr>
</table>

#### Table Content Evaluation (ROUGE-L Score)

<table>
  <tr>
    <th rowspan="2">Model</th>
    <th rowspan="2">Overall</th>
    <th colspan="3">Difficulty</th>
    <th colspan="2">Category</th>
    <th colspan="2">Source Type</th>
  </tr>
  <tr>
    <th>Hard</th>
    <th>Easy</th>
    <th>Medium</th>
    <th>Fixed Header</th>
    <th>Open Header</th>
    <th>Generate</th>
    <th>Retrieve</th>
  </tr>
  <tr>
    <td>ODIE-8B-4Bit-Llama3.1-inst</td>
    <td>50.14</td>
    <td>51.01</td>
    <td>51.33</td>
    <td>48.30</td>
    <td>51.41</td>
    <td>46.11</td>
    <td>54.87</td>
    <td>48.90</td>
  </tr>
</table>

## Reported results in the paper
![image](https://github.com/user-attachments/assets/a4bee4a1-e626-47fe-8714-2a95e731717d)

### Model and Training Configuration

- **Base Model:** Meta-Llama-3.1-8B-Instruct
- **Data Path:** `data/training_data.jsonl`
- **Output Directory:** `odie-7b`
- **Epochs:** 3
- **Batch Size:** 16
- **Micro Batch Size:** 4
- **Learning Rate:** 3e-4
- **Cutoff Length:** 2048
- **Validation Set Size:** 0
- **LoRA Configuration:**
  - `lora_r`: 16
  - `lora_alpha`: 16
  - `lora_dropout`: 0.05
  - `lora_target_modules`: ["q_proj", "k_proj", "v_proj", "o_proj", "up_proj", "down_proj", "gate_proj", "embed_tokens", "lm_head"]
- **Other Hyperparameters:**
  - `train_on_inputs`: False
  - `add_eos_token`: False
  - `group_by_length`: False
  - `use_chat_prompt`: True
  - `prompt_template_name`: "alpaca"

### Quantization and Device Configuration

- **BitsAndBytes Configuration:**
  - `load_in_4bit`: True
  - `bnb_4bit_use_double_quant`: True
  - `bnb_4bit_quant_type`: "nf4"
  - `bnb_4bit_compute_dtype`: torch.bfloat16
- **Device Map:** Auto
- **Distributed Data Parallel (DDP):** Enabled (2 GPUs)

**Comment by D0ub1ePieR** [comment]
From Table 3 you can see that alpaca used 52K instruct tunning data, while ODIE only used 7.5K data. It seems to me that the model used should be the llama-7B-base model rather than the instruct model ?

Also the model used in train.sh is elinas/llama-7b-hf-transformers-4.29, which should also be a model that has not been fine-tuned with instruct tuning.

**Comment by JulianOestreich90** [comment]
Hey Abdallah, 
Did you used the exact same code for finetuning and inference as provided here in this repository or would you mind providing your code as well?

**Comment by abdallaharioua** [comment]
Hello,
The exact same code.

**Comment by JulianOestreich90** [comment]
And for inference? The inference is incredible slow with the provided code (~30min for 150 samples)
